sim/pokemon.js

```diff
@@ -966,7 +966,12 @@ class Pokemon {
 						this.battle.add('-burst', this, apparentSpecies, template.requiredItem);
 						this.moveThisTurnResult = true; // Ultra Burst counts as an action for Truant
 					} else if (source.onPrimal) {
-						this.battle.add('-primal', !this.illusion && this);
+						if (this.illusion) {
+							this.ability = '';
```

Comment:
Also why is the ability cleared here?

Reply:
Oh, nvm, I see.

Reply:
So the illusion remains up while the Pokemon is still in first-turn and isn't immediately removed upon reverting, similarly to mega evolution. The Desolate Land/Primordial Sea still activates and remains up until switching as normal.

Reply:
oh